checkup.go

```diff
@@ -359,6 +359,12 @@ type Storage interface {
 	Store([]Result) error
 }
 
+// StorageReader can read results from the Storage.
+type StorageReader interface {
+	Fetch(string) ([]Result, error)
+	GetIndex() (map[string]int64, error)
```

Comment:
You're right @shurcooL, thanks for the comment.
I'll submit a patch for this.

BTW, I'm not sure the map details are useful to the user: I only use the keys and I guess the values might be backend-specific (fs & github use int64, s3 has no index AFAICS).

Maybe the map should be a map[string]interface{}, or maybe a list of keys should be returned instead?
@mholt, any thought on this?